R/tm_g_ae_sub.R

```diff
@@ -289,40 +285,41 @@ srv_g_ae_sub <- function(id,
     output_q <- reactive({
       ANL <- data[[dataname]]() # nolint
       ADSL <- data[["ADSL"]]() # nolint
-      validate_has_data(ANL, min_nrow = 10)
-      iv_comp <- shinyvalidate::InputValidator$new()
-      iv_comp$add_rule("arm_trt", shinyvalidate::sv_not_equal(
+
+      teal::validate_has_data(ANL, min_nrow = 10, msg = sprintf("%s has not enough data", dataname))
+
+      iv <- shinyvalidate::InputValidator$new()
+      iv$add_rule("arm_var", shinyvalidate::sv_required(
+        message = "Arm Variable is required"
+      ))
+      iv$add_rule("arm_var", ~ if (!is.factor(ANL[[req(.)]])) {
+        "Arm Var must be a factor variable, contact developer"
+      })
+      iv$add_rule("arm_trt", shinyvalidate::sv_not_equal(
         input$arm_ref,
-        message_fmt = "Must not be equal to Control"
+        message_fmt = "Control and Treatment must be different"
       ))
-      iv_comp$add_rule("arm_ref", shinyvalidate::sv_not_equal(
+      iv$add_rule("arm_ref", shinyvalidate::sv_not_equal(
         input$arm_trt,
-        message_fmt = "Must not be equal to Treatment"
+        message_fmt = "Control and Treatment must be different"
+      ))
+      iv$add_rule("groups", shinyvalidate::sv_in_set(
+        names(ANL),
+        message_fmt = "Groups must be a variable in ANL"
```

Comment:
There is no such dataset as ANL for the end user. `ANL <- data[[dataname]]()`

```suggestion
        message_fmt = sprintf("Groups must be a variable in %s", dataname)
```

Reply:
"ADAE" then? See line 321 in original version.